### ISSUE TITLE
Pricebook modifiers are not working on Quote Items

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#3709]
**[Mingle Card: 4031](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/4031)**
Steps to reproduce
------------------

```
| | |
|-|-|
|**Version #**|
|Master|
|**OS**|
| |
|**Browser**|
| |
|**Username**|
|ajf@t3.com|
|**Password**|
|computer45|
  
 

```

1.  Create a quote with ‘Price Book Modifier PB’ price book
2.  Add a Locked price item
3.  Open the item and edit the price

Expected result
---------------

Price field should be read only

Actual Result
-------------

Price field is editable and edited price is displayed on the quote item grid.

NOTES:
------

Original PR:

[https://github.com/Liquidframeworks/alpine-mobile/pull/1890](https://github.com/Liquidframeworks/alpine-mobile/pull/1890)

New PR:

[https://github.com/Liquidframeworks/alpine-mobile/pull/1981](https://github.com/Liquidframeworks/alpine-mobile/pull/1981)

Analysis
--------

Yup.  Not doing anything with the PB Modifier fields.  

Talking with DP about another card, he has implemented 2 new features to the “layout/form-edit” bindings (one is already merged, the other should go in hopefully soon).  He added skipFields and lockFields;

*   skipFields does not display the field on the form at all, while
*   getLockedFields simply makes the “PBI\_Locked\_\*” fields readonly (not merged yet, [https://github.com/dpitre/alpine-mobile/tree/4116\_incremental\_ticket\_items](https://github.com/dpitre/alpine-mobile/tree/4116_incremental_ticket_items))

We can either wait for DP’s branch/PR to merge (+1 week after 4100 is merged) or we can implement it as part of this card. (TOM SAYS TO WAIT)

UPDATE: BR – 6/24/2016: 

We need to leverage the quote-item-override filter from card \[alpine\_mobile/#4085\] Grid editing quote: some fields on the grid should be locked for editing.

Related Cards
-------------

\[alpine\_mobile/#4085\] Grid editing quote: some fields on the grid should be locked for editing

Dependent on PR [https://github.com/Liquidframeworks/alpine-mobile/pull/1889](https://github.com/Liquidframeworks/alpine-mobile/pull/1889)

\[alpine\_mobile/#4218\] Use raw layouts and description for ticket item details #1905

Test Plan
---------

Note that this card is dependent on card #4085 and PR 1889 being merged first.

This card was fixed as part of \[alpine\_mobile/#4218\] Use raw layouts and description for ticket item details  (already merged via PR #1905)

Selenium > pbmodifiersQuoteItems.js

### Checklist for Merging

*   Enter the Pull Request number in the card properties
*   How is this tested in Selenium? 
*   Do we have adequate unit testing? 
*   Where is this covered in documentation?
*   Where is this in Tarantula?

**Comment by githubbob42** [comment]
Matthew Todd : 05/31/2016 10:23 CDT
Reproduced with 4.0.1.3243

**Comment by githubbob42** [comment]
Matthew Todd : 05/31/2016 10:31 CDT
Reproduced with 4.0.1.3243

**Comment by githubbob42** [comment]
**Pull Request #1981**
